code/modules/food_and_drink/candy.dm

```diff
@@ -137,88 +137,98 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/candy/jellybean)
 	name = "\improper Farty Snott's Every Flavour Bean"
 	desc = "A favorite halloween sweet worldwide!"
-
+	// sugar content is less than the flavouring content, so that wizard golems are guaranteed to be named after the flavouring
+	sugar_content = 40
 	New()
 		..()
 		SPAWN(0)
-			if (src.reagents)
-				if (prob(12))
-					src.reagents.add_reagent(pick("milk", "coffee", "VHFCS", "gravy", "fakecheese", "grease"), 10)
-					src.heal_amt = 1
-				else if (prob(12))
-					src.reagents.add_reagent(pick("bilk", "beff", "vomit", "gvomit", "porktonium", "badgrease"), 10)
-					src.heal_amt = 0
+			if (!src.reagents) return
+			var/reagent = null
+			if (prob(12))
+				reagent = pick("milk", "coffee", "VHFCS", "gravy", "fakecheese", "grease")
+				src.heal_amt = 1
+			else if (prob(12))
+				reagent = pick("bilk", "beff", "vomit", "gvomit", "porktonium", "badgrease")
+				src.heal_amt = 0
+			else
+				if (length(all_functional_reagent_ids) > 0)
+					reagent = pick(all_functional_reagent_ids)
 				else
-					var/flavor = null
-					if (length(all_functional_reagent_ids) > 0)
-						flavor = pick(all_functional_reagent_ids)
-					else
-						flavor = "sugar"
-					src.reagents.add_reagent(flavor, 50)
-
-				src.food_color = src.reagents.get_master_color()
+					reagent = "sugar"
+
+			var/datum/reagent/current = reagents_cache[reagent]
+			if (current)
+				// make space for the flavouring if there is none
+				if(src.reagents.maximum_volume - src.reagents.total_volume < 50)
+					src.reagents.maximum_volume += 50
+				src.reagents.add_reagent(reagent, 50)
+				src.food_color = rgb(current.fluid_r, current.fluid_g, current.fluid_b, max(current.transparency,255))
 				src.icon += src.food_color // apparently this is a thing you can do?  neat!
 
-				src.reagents.add_reagent("sugar", 50)
-				if (src.reagents.total_volume <= 60)
-					src.reagents.add_reagent("sugar", 40)
+			if(prob(50))
+				flavor = pick("egg", "vomit", "snot", "poo", "earwax", "wet dog", "belly-button lint", "sweat", "congealed farts", "mold", "armpits", "elbow grease", "sour milk", "WD-40", "slime", "blob", "gym sock", "pants", "brussels sprouts", "feet", "litter box", "durian fruit", "asbestos", "corpse flower", "corpse", "cow dung", "rot", "tar", "ham", "gooncode", "quark-gluon plasma", "bee", "heat death")
+				phrase = pick("Oh god", "Jeez", "Ugh", "Blecch", "Holy crap that's awful", "What the hell?", "*HURP*", "Phoo")
+				tastesbad = 1
+			else
+				flavor = pick("egg", "strawberry", "raspberry", "snozzberry", "happiness", "popcorn", "buttered popcorn", "cinnamon", "macaroni and cheese", "pepperoni", "cheese", "lasagna", "pina colada", "tutti frutti", "lemon", "margarita", "coconut", "pineapple", "scotch", "vodka", "root beer", "cotton candy", "Lagavulin 18", "toffee", "vanilla", "coffee", "apple pie", "neapolitan", "orange", "lime", "mango", "apple", "grape", "Slurm")
+				phrase = pick("Yum", "Wow", "MMM", "Delicious", "Scrumptious", "Fantastic", "Oh yeah")
```

Comment:
Maybe add flavor and phrase list vars on the jellybean parent so you can just do `pick(bad_flavors)` / `pick(good_flavors)` or whatever instead of duplicating this stuff on both the `someflavor` and `everyflavor` children? I think they're meant to contain the exact same strings.

Reply:
You're right it looks like they're pretty much identical, I've refactored as you suggest. I also removed the need for the flavor/phrase variables for the sake of optimisation, so now beans have Schrodinger's flavour until eaten. I don't think that makes any practical difference unless admins were peaking on jellybean flavour-flavourtext prior to eating them for some reason.

Reply:
Oh, I guess it does prevent admins from editing flavourtext mid-round, maybe that's something worth keeping?

Reply:
This seems like a super niche thing that I've never seen get edited so I think it's probably fine to remove it in favor of better code!